src/Castle.Core/DynamicProxy/CustomAttributeInfo.cs

```diff
@@ -293,20 +295,20 @@ private static int CombineMemberHashCodes(IDictionary<string, object> dict)
 			}
 		}
 
-		private IDictionary<string, object> MakeNameValueDictionary<T>(T[] members, object[] values)
+		private IDictionary<string, object?> MakeNameValueDictionary<T>(T[] members, object?[] values)
 			where T : MemberInfo
 		{
-			var dict = new Dictionary<string, object>();
+			var dict = new Dictionary<string, object?>();
 			for (int i = 0; i < members.Length; i++)
 			{
 				dict.Add(members[i].Name, values[i]);
 			}
 			return dict;
 		}
 
-		private class AttributeArgumentValueEqualityComparer : IEqualityComparer<object>
+		private class AttributeArgumentValueEqualityComparer : IEqualityComparer<object?>
 		{
-			bool IEqualityComparer<object>.Equals(object x, object y)
+			new public bool Equals(object? x, object? y)
```

Comment:
Does this need the `new` keyword?

Reply:
I'd say so, because without it we get the following compiler warning:

> `CustomAttributeInfo.AttributeArgumentValueEqualityComparer.Equals(object?, object?)` hides inherited member `object.Equals(object, object)`. Use the `new` keyword if hiding was intended.

(It's actually interesting that `object.Equals` appears to have incorrect nullability hints for its parameters.)